plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -147,6 +148,23 @@ export class SessionRecordingIngester {
     ) {
         this.isDebugLoggingEnabled = buildIntegerMatcher(config.SESSION_RECORDING_DEBUG_PARTITION, true)
 
+        // Parse SESSION_RECORDING_V2_METADATA_SWITCHOVER as ISO datetime
+        if (config.SESSION_RECORDING_V2_METADATA_SWITCHOVER) {
+            const parsed = Date.parse(config.SESSION_RECORDING_V2_METADATA_SWITCHOVER)
+            if (!isNaN(parsed)) {
+                this.metadataSwitchoverDate = new Date(parsed)
+                logger.info('SESSION_RECORDING_V2_METADATA_SWITCHOVER enabled', {
+                    value: config.SESSION_RECORDING_V2_METADATA_SWITCHOVER,
+                    parsedDate: this.metadataSwitchoverDate.toISOString(),
+                })
+            } else {
+                logger.warn('SESSION_RECORDING_V2_METADATA_SWITCHOVER is not a valid ISO datetime', {
```

Comment:
Do we _maybe_ want to throw here instead? Like if the value is set but it can't be parsed maybe its better we crash out